### PR TITLE
Plugins: Use plugins SDK constants for OAuth header names

### PR DESCRIPTION
**Pull request by wbrowne**

### CODE REVIEW

**Review by svennergr — APPROVED** [review]
Just to be sure, headers here should already be canonicalized, right?

`idTokenHeaderName` is `X-ID-Token` and `OAuthIdentityIDTokenHeaderName` is `X-Id-Token`. So there's a difference, but I assume both should work - `X-Id-Token` should even be more correct.

**Comment by wbrowne** [comment]
> Just to be sure, headers here should already be canonicalized, right?
> 
> `idTokenHeaderName` is `X-ID-Token` and `OAuthIdentityIDTokenHeaderName` is `X-Id-Token`. So there's a difference, but I assume both should work - `X-Id-Token` should even be more correct.

Yeah I think this is "more correct" as you say since we're using `textproto.CanonicalMIMEHeaderKey(k)` when comparing against `OAuthIdentityIDTokenHeaderName` in the plugin.

File: pkg/services/pluginsintegration/clientmiddleware/oauthtoken_middleware.go [hunk 1]
```diff
@@ -69,19 +64,19 @@ func (m *OAuthTokenMiddleware) applyToken(ctx context.Context, pCtx backend.Plug
 
 			switch t := req.(type) {
 			case *backend.QueryDataRequest:
-				t.Headers[tokenHeaderName] = authorizationHeader
+				t.Headers[backend.OAuthIdentityTokenHeaderName] = authorizationHeader
 				if idTokenHeader != "" {
-					t.Headers[idTokenHeaderName] = idTokenHeader
+					t.Headers[backend.OAuthIdentityIDTokenHeaderName] = idTokenHeader
```
Comment (xnyo) [inline pkg/services/pluginsintegration/clientmiddleware/oauthtoken_middleware.go:69]: `backend.OAuthIdentityIDTokenHeaderName` seems to be `X-Id-Token`

https://github.com/grafana/grafana-plugin-sdk-go/blob/c5aa822be58253fd954e921aec00241833d9ca4c/backend/http_headers.go#L17

but the `idTokenHeaderName` from the removed const was `X-ID-Token`

(capital "D" vs lowercase "d")

Are we sure this isn't going to cause any issues?

**Review by xnyo — APPROVED** [review]
LGTM but like Sven I am a bit concerned about `X-ID-Token` vs `X-Id-Token`

File: pkg/services/pluginsintegration/clientmiddleware/oauthtoken_middleware.go (hunk 1 above)
Comment (wbrowne) [reply pkg/services/pluginsintegration/clientmiddleware/oauthtoken_middleware.go:69]: I don't _believe_ so, but I will hold off merging this before confirming myself. There's no rush on this and can also probably wait for the expert in this area @marefr!

Comment (wbrowne) [reply pkg/services/pluginsintegration/clientmiddleware/oauthtoken_middleware.go:69]: Had a quick look into this. I think all is fine

```go
const OAuthIdentityIDTokenHeaderName = "X-Id-Token"

func getHTTPHeadersFromStringMap(headers map[string]string) http.Header {
	httpHeaders := http.Header{}

	for k, v := range headers {
         // when k == "X-ID-Token" 
         // then textproto.CanonicalMIMEHeaderKey(k) == "X-Id-Token"
		if textproto.CanonicalMIMEHeaderKey(k) == OAuthIdentityIDTokenHeaderName {
			httpHeaders.Set(k, v)
		}
//...
```

Comment (marefr) [reply pkg/services/pluginsintegration/clientmiddleware/oauthtoken_middleware.go:69]: We're already referring to them in https://grafana.com/developers/plugin-tools/how-to-guides/data-source-plugins/add-authentication-for-data-source-plugins#extract-a-header-from-an-http-request so should be fine I think

**Review by marefr — APPROVED** [review]
LGTM

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**